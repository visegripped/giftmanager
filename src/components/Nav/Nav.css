.nav-list {
  list-style: none;
  padding: 0;
  margin: 0;
}

.nav-item {
  display: inline-block;
  margin:0;
  padding:0;
}

.nav-item > a {
  background-color: var(--backgroundColorContent);
  padding: 0 1vw;
  margin: 0 1vw;
  border: 1px solid #ccc;
  border-bottom-width: 0;
  text-decoration: none;
  display: block;
  border-radius: .5vh;
  border-bottom-right-radius: 0;
  border-bottom-left-radius: 0;
  bottom: -1px;
  position: relative;
  font-size: 1.5vh;
}