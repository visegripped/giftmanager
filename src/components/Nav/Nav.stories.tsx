import React from 'react';
import { ComponentStory, ComponentMeta } from '@storybook/react';
import { BrowserRouter } from 'react-router-dom';
import Nav from './index';

export default {
  title: 'component/Nav',
  component: Nav,
} as ComponentMeta<typeof Nav>;

const Template: ComponentStory<typeof Nav> = (args) => (
  <BrowserRouter>
    <Nav {...args} />
  </BrowserRouter>
);

export const BasicUsage = Template.bind({});
BasicUsage.args = {
  cssClasses: 'test',
};
