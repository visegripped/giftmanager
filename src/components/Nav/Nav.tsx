import React from 'react';
import { Link } from 'react-router-dom';
import './Nav.css';

interface NavProps {
  cssClasses?: string;
}

const Nav = (props: NavProps) => {
  const { cssClasses } = props;
  const className = `${cssClasses}`;
  return (
    <nav className={className} data-testid="Nav">
      <ul className="nav-list">
        <li className="nav-item">
          <Link to="/mylist">My list</Link>
        </li>
        <li className="nav-item">
          <Link to="/theirlist">Their list</Link>
        </li>
      </ul>
    </nav>
  );
};

export default Nav;
