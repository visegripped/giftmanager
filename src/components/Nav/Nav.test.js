import { render, screen } from '@testing-library/react';
import Nav from './index';
import { BrowserRouter } from 'react-router-dom';

describe('Nav', () => {
  test('snapshot matches', () => {
    render(<Nav cssClasses="test classes" />, { wrapper: BrowserRouter });
    const renderedNav = screen.getByTestId('Nav');
    expect(renderedNav).toMatchSnapshot();
  });
});
