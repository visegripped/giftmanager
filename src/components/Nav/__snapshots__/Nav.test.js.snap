// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav snapshot matches 1`] = `
<nav
  class="test classes"
  data-testid="Nav"
>
  <ul
    class="nav-list"
  >
    <li
      class="nav-item"
    >
      <a
        href="/mylist"
      >
        My list
      </a>
    </li>
    <li
      class="nav-item"
    >
      <a
        href="/theirlist"
      >
        Their list
      </a>
      <div
        class="nav-menu-container nav-menu-container--hidden"
      >
        <menu
          data-testid="Menu"
        />
      </div>
    </li>
  </ul>
</nav>
`;
